webapp/channels/src/components/team_groups_manage_modal/team_groups_manage_modal.test.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {shallow} from 'enzyme';
+import React from 'react';
+
+import TeamGroupsManageModal from 'components/team_groups_manage_modal/team_groups_manage_modal';
+
+import {TestHelper} from 'utils/test_helper';
+import { shallowWithIntl } from 'tests/helpers/intl-test-helper';
+
+describe('components/TeamGroupsManageModal', () => {
+    const team = TestHelper.getTeamMock({ type: 'O', allowed_domains: '', allow_open_invite: false, scheme_id: undefined });
+    const actions = {
+        getGroupsAssociatedToTeam: jest.fn().mockResolvedValue({data: {groups: [], totalGroupCount: 0}}),
+        closeModal: jest.fn(),
+        openModal: jest.fn(),
+        unlinkGroupSyncable: jest.fn().mockResolvedValue({data: []}),
+        patchGroupSyncable: jest.fn().mockResolvedValue({data: []}),
+        getMyTeamMembers: jest.fn(),
+    };
+
+    const baseProps = {
+        intl: {
+            formatMessage: jest.fn(),
+        },
+        team,
+        actions
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(<TeamGroupsManageModal {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should hide confirm modal when team changes', () => {
+        const wrapper = shallow(<TeamGroupsManageModal {...baseProps}/>);
+        wrapper.setProps({team: TestHelper.getTeamMock({id: 'new'})});
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should call getGroupsAssociatedToTeam on loadItems', async () => {
+        const wrapper = shallowWithIntl(<TeamGroupsManageModal {...baseProps}/>);
+        const instance = wrapper.instance() as any;
```

Comment:
```suggestion
        const instance = wrapper.instance() as TeamGroupsManageModal;
```

Reply:
Hey sure. I'll keep this in my mind while making the updated test cases for checking the types instead of any to the respective component type in case of a component.